pkg/xds/envoy/clusters/v2/circuit_breaker_configurer.go

```diff
@@ -3,86 +3,34 @@ package clusters
-		outlierDetection.ConsecutiveGatewayFailure = gateway.GetConsecutive()
-		outlierDetection.EnforcingConsecutiveGatewayFailure = &wrappers.UInt32Value{Value: 100}
-	} else {
-		outlierDetection.EnforcingConsecutiveGatewayFailure = &wrappers.UInt32Value{Value: 0}
-	}
-}
-
-func (c *OutlierDetectionConfigurer) configureLocalErrorDetector(outlierDetection *envoy_cluster.OutlierDetection) {
-	if local := c.CircuitBreaker.Spec.GetConf().GetDetectors().GetLocalErrors(); local != nil {
-		outlierDetection.ConsecutiveLocalOriginFailure = local.GetConsecutive()
-		outlierDetection.EnforcingConsecutiveLocalOriginFailure = &wrappers.UInt32Value{Value: 100}
-	} else {
-		outlierDetection.EnforcingConsecutiveLocalOriginFailure = &wrappers.UInt32Value{Value: 0}
+	if !c.CircuitBreaker.HasThresholds() {
+		return nil
 	}
-}
-
-func (c *OutlierDetectionConfigurer) configureStandardDeviationDetector(outlierDetection *envoy_cluster.OutlierDetection) {
-	if stdev := c.CircuitBreaker.Spec.GetConf().GetDetectors().GetStandardDeviation(); stdev != nil {
-		outlierDetection.SuccessRateRequestVolume = stdev.GetRequestVolume()
-		outlierDetection.SuccessRateMinimumHosts = stdev.GetMinimumHosts()
-		if factor := stdev.GetFactor(); factor != nil {
-			outlierDetection.SuccessRateStdevFactor = &wrappers.UInt32Value{Value: uint32(factor.GetValue() * 1000)}
-		}
-		outlierDetection.EnforcingSuccessRate = &wrappers.UInt32Value{Value: 100}
-		outlierDetection.EnforcingLocalOriginSuccessRate = &wrappers.UInt32Value{Value: 100} // takes effect only when SplitExternalLocalOriginErrors is true
-	} else {
-		outlierDetection.EnforcingSuccessRate = &wrappers.UInt32Value{Value: 0}
-	}
-}
-
-func (c *OutlierDetectionConfigurer) configureFailureDetector(outlierDetection *envoy_cluster.OutlierDetection) {
-	if failure := c.CircuitBreaker.Spec.GetConf().GetDetectors().GetFailure(); failure != nil {
-		outlierDetection.FailurePercentageRequestVolume = failure.GetRequestVolume()
-		outlierDetection.FailurePercentageMinimumHosts = failure.GetMinimumHosts()
-		outlierDetection.FailurePercentageThreshold = failure.GetThreshold()
-
-		outlierDetection.EnforcingFailurePercentage = &wrappers.UInt32Value{Value: 100}
-		outlierDetection.EnforcingFailurePercentageLocalOrigin = &wrappers.UInt32Value{Value: 100} // takes effect only when SplitExternalLocalOriginErrors is true
-	} else {
-		outlierDetection.EnforcingFailurePercentage = &wrappers.UInt32Value{Value: 0}
+	cluster.CircuitBreakers = &envoy_cluster.CircuitBreakers{
+		Thresholds: []*envoy_cluster.CircuitBreakers_Thresholds{
+			{
+				Priority:           envoy_api_core.RoutingPriority_DEFAULT,
+				MaxConnections:     c.CircuitBreaker.Spec.Conf.GetThresholds().GetMaxConnections(),
+				MaxPendingRequests: c.CircuitBreaker.Spec.Conf.GetThresholds().GetMaxPendingRequests(),
+				MaxRetries:         c.CircuitBreaker.Spec.Conf.GetThresholds().GetMaxRetries(),
+				MaxRequests:        c.CircuitBreaker.Spec.Conf.GetThresholds().GetMaxRequests(),
```

Comment:
1. What if `c.CircuitBreaker.Spec.Conf.GetThresholds()` would return nil?
2. Maybe we could DRY `c.CircuitBreaker.Spec.Conf.GetThresholds()`

Reply:
1. same, I don't think we need nil-checks here
2. I don't mind to have a variable `thresholds`, fixed